kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -901,24 +919,33 @@ class ResidualBasedNewtonRaphsonStrategy
         mpConvergenceCriteria->InitializeNonLinearIteration(r_model_part, r_dof_set, rA, rDx, rb);
         bool is_converged = mpConvergenceCriteria->PreCriteria(r_model_part, r_dof_set, rA, rDx, rb);
 
+
         // Function to perform the building and the solving phase.
-        if (BaseType::mRebuildLevel > 0 || BaseType::mStiffnessMatrixIsBuilt == false) {
-            TSparseSpace::SetToZero(rA);
-            TSparseSpace::SetToZero(rDx);
+        if ((GetUseInitialStiffnessFlag() && r_model_part.GetProcessInfo()[STEP] == 1)) {
```

Comment:
the checko on STEP is fragile ... it is better to rely on mStiffnessMatrixIsBuilt.